posts/2021-03-03-jakarta-ee-9-compatibility.adoc

```diff
@@ -0,0 +1,26 @@
+title: Open Liberty beta is Jakarta EE 9 compatible
+categories: blog
+author_picture: https://avatars3.githubusercontent.com/jhanders34
+author_github: https://github.com/jhanders34
+seo-title: Open Liberty beta is Jakarta EE 9 compatible - OpenLiberty.io
+seo-description: Open Liberty beta releases have satisified the Jakarta EE 9 TCK and have been added to the Jakarta EE website.
+blog_description: "Open Liberty beta releases have satisified the Jakarta EE 9 TCK and have been added to the Jakarta EE website"
+open-graph-image: https://openliberty.io/img/twitter_card.jpg
+---
+= Open Liberty beta is Jakarta EE 9 compatible
+Jared Anderson <https://github.com/jhanders34>
+:imagesdir: /
+:url-prefix: https://openliberty.io
+:url-about: /
+//Blank line here is necessary before starting the body of the post.
+
+Late last year, the Jakarta EE community released the next step in the evolution of the Jakarta EE specification.  link:https://jakarta.ee/specifications/platform/9/jakarta-platform-spec-9.html[Version 9.0 of the Jakarta EE specification] focuses on renaming packages and namespaces by changing to jakarta prefixed packages names in the API and to the link:https://jakarta.ee/xml/ns/jakartaee/[jakarta.ee namespace] in the deployment descriptor files.  Additionally, the specification makes optional or removes older functions that have been deprecated or stabilized for many years.
+
+The Open Liberty development team has been working over the last year to implement the Jakarta EE 9 specification as it was being finalized.  In an effort to not have to duplicate the same code with references to different API package names, the team developed an link:https://projects.eclipse.org/projects/technology.transformer[Open Source byte code transformation technology].  This technology allows for the same code to be compiled to be Jakarta EE 8 compatible and then be transformed to work with Jakarta EE 9 by manipulating the byte code to convert class and property references from being prefixed with javax to jakarta.
+
+As of the link:{url-prefix}/blog/2021/01/26/ee9-messaging-security-21002-beta.html[21.0.0.2-beta release], Open Liberty is the first vendor product added to the link:https://jakarta.ee/compatibility/#tab-9[Jakarta EE website] to be Jakarta EE Web Profile 9.0 compatible.  With the recent link:{url-prefix}/blog/2021/02/19/mpcontextpropagation-requesttiming-21003-beta.html[21.0.0.3-beta release], Open Liberty is the first vendor product to be added to the Jakarta EE Platform 9.0 compatibility list.  These releases continue to demonstrate Open Liberty's leadership in the Jakarta EE technologies.  With the upcoming link:https://jakarta.ee/specifications/platform/9.1/[Jakarta EE 9.1 release] that adds Java 11 support to the TCK, I anticipate that Open Liberty will be added to the compatibility list for the 9.1 release when it is available since link:{url-prefix}/blog/2019/02/06/java-11.html[Open Liberty has supported Java 11] for the last 2 years.
```

Comment:
_Open Liberty is the first vendor product added to the [Jakarta EE website] to be Jakarta EE Web Profile 9.0 compatible_

is _added to the Jakarta EE website_ an important clarification? If not, I would just say "...the first vendor product to be Jakarta EE Web Profile 9.0 compatible" and hyperlink that.

Reply:
Updated